yarn-project/acir-simulator/src/avm/avm_execution_environment.ts

```diff
@@ -36,4 +37,58 @@ export class AvmExecutionEnvironment {
     /** - */
     public readonly calldata: Fr[],
   ) {}
+
+  public newCall(address: AztecAddress): AvmExecutionEnvironment {
+    return new AvmExecutionEnvironment(
+      address,
+      address,
+      this.origin,
+      this.sender,
+      this.portal,
+      this.feePerL1Gas,
+      this.feePerL2Gas,
+      this.feePerDaGas,
+      this.contractCallDepth,
+      this.globals,
+      this.isStaticCall,
+      this.isDelegateCall,
+      this.calldata,
+    );
+  }
+
+  public newStaticCall(address: AztecAddress): AvmExecutionEnvironment {
+    return new AvmExecutionEnvironment(
+      address,
+      address,
+      this.origin,
+      this.sender,
+      this.portal,
+      this.feePerL1Gas,
+      this.feePerL2Gas,
+      this.feePerDaGas,
+      this.contractCallDepth,
+      this.globals,
+      true,
```

Comment:
Could you add /*paramName=*/? (In general I think it's a good practice when not clear from context)